### PR TITLE
feat: walkthrough optional footer

### LINKED ISSUE #27335: [Feature]: Multiple Environment Walkthrough/Exploration
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Implement walkthrough to improve ME discoverability as per the [designs](https://www.figma.com/file/QbU4HFVRLpWtbQQIcEFEca/Multiple-Environments-Phase-2?type=design&node-id=398-12681&mode=design&t=VmgjjgrGG3omZvB0-4)

Notion [link](https://www.notion.so/appsmith/Environments-Phase-2-9cc5f1153fe94dcf8fe86e6ede9931b6#f945dc70b8b547979e1789abb9bd0b99)

### Why should this be worked on?

Users are not able to explore new ME feature as per expectation. Hence this walkthrough will help user to get introduced to ME first time.

### PR DESCRIPTION
**Pull request by ayushpahwa**
## Description
This PR allows to add a footer for any walkthrough.

#### PR fixes following issue(s)
Fixes #27335 

#### Type of change

- New feature (non-breaking change which adds functionality)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by ayushpahwa** [comment]
Adding test plan label since this PR just adds a new feature for walkthrough functionality. Usage will be added via EE codebase

File: app/client/src/components/featureWalkthrough/walkthroughRenderer.tsx [hunk 1]
```diff
@@ -295,6 +308,17 @@ const InstructionsComponent = ({
           <img src={details.imageURL} />
         </ImageWrapper>
       )}
+      {details.showFooterButton && !!details.footerDetails && (
```
Comment (AmanAgarwal041) [inline app/client/src/components/featureWalkthrough/walkthroughRenderer.tsx:311]: Does it make sense to add `showFooterButton` as when the `footerDetails` value is set then we will show the walkthrough footer else not?

**Review by AmanAgarwal041 — APPROVED** [review]

File: app/client/src/components/featureWalkthrough/walkthroughContext.tsx [hunk 2]
```diff
@@ -27,8 +27,25 @@ export interface FeatureDetails {
   description: string;
   // Gif or Image to give a walkthrough
   imageURL?: string;
+  // footer details
+  footerDetails?: FeatureFooterDeatils;
 }
 
+export interface FeatureFooterDeatils {
```
Comment (dipyamanbiswas07) [inline app/client/src/components/featureWalkthrough/walkthroughContext.tsx:34]: Please fix the spelling

Comment (ayushpahwa) [reply app/client/src/components/featureWalkthrough/walkthroughContext.tsx]: Damn, my bad. Fixed it now.

**Review by dipyamanbiswas07 — APPROVED** [review]

**Review by dipyamanbiswas07 — APPROVED** [review]

**Pull request merged**